### PR TITLE
Improve test app, add top-level option "issuer"

### PR DESCRIPTION
**Pull request by aarongranick-okta**
## Description:
- adds top-level configuration option "issuer". If issuer is provided, "baseUrl" is optional.
- improves typescript test app (yarn start:test:app)
  - adds typescript linting
  - adds form for setting option
  - puts config in URL
  - adds signOut and clear storage buttons
  - adds configured flow as drop down option

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:

Before:

Screen Shot 2021-12-15 at 3 24 05 PM

Now:

Screen Shot 2021-12-15 at 9 44 04 AM

### Reviewers:
@shuowu-okta 

### Issue:

- [OKTA-454844](https://oktainc.atlassian.net/browse/OKTA-454844)

### CODE REVIEW

File: README.md
Comment (pradeepdewda-okta) [inline README.md:776]: removing baseUrl entirely ? 

can't be `baseUrl == issuer` ?

File: src/models/Settings.js
Comment (pradeepdewda-okta) [inline src/models/Settings.js:371]: this can be merged to the `if` above right ?

**Review by pradeepdewda-okta — APPROVED** [review]
Looks good to me

File: README.md
Comment (aarongranick-okta) [reply README.md:776]: Yes, it can be removed. I was in the process of updating the README, but decided to hold off as this will be merged to master before the change is pushed. There can be a README update after this version is publicly released

File: src/models/Settings.js
Comment (aarongranick-okta) [reply src/models/Settings.js:371]: It could... but the block above sets the baseUrl if it is not set, inferring value from issuer. I thought this was a distinct logical block from throwing the error.

**Review by shuowu-okta — APPROVED** [review]

File: src/models/Settings.js
Comment (aarongranick-okta) [reply src/models/Settings.js:371]: I was thinking "setting defaults" vs. "asserting required", but maybe it does make more sense to do it by parameter name.

**Pull request closed without merging**
